### PR TITLE
Pending Trips Frontend Part 2: Accept and Reject Trips

### PR DESCRIPTION
**Pull request by zghera**
### What is a quick description of the change?

Add functionality to accept and reject trips from the pending trip view as well as "re-"accept trips from the rejected trip view.

### Is this fixing an issue?

none

### Are there more details that are relevant?

none

### Check lists (check `x` in `[ ]` of list items)

- [  ] Test written/updating
- [  ] Tests passing
- [x] Coding style (indentation, etc)

Only UI based changes.

Same manual tests as in #123 .

### Any additional comments?

{Fill in here or remove (leave question)}
Untitled_ Aug 5, 2020 11_09 PM (1)

### CODE REVIEW

File: frontend/src/components/ViewTrips/change-collaborator-type-button.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    curCollabUidSet.delete(curUserUid);
+    let curCollabUidArr = Array.from(curCollabUidSet);
+
+    let newCollabUidSet = new Set(props.tripData[props.newCollabField]);
+    newCollabUidSet.add(curUserUid);
+    let newCollabUidArr = Array.from(newCollabUidSet);
+
+    return { [props.curCollabField]: curCollabUidArr,
+             [props.newCollabField]: newCollabUidArr
+           };
```
Comment (ananyeet) [inline frontend/src/components/ViewTrips/change-collaborator-type-button.js:47]:
```suggestion
    return { 
      [props.curCollabField]: curCollabUidArr,
      [props.newCollabField]: newCollabUidArr
    };
```

jsdocs

File: frontend/src/components/ViewTrips/trip.js [hunk 2]
```diff
@@ -68,6 +74,8 @@ function getTripInfoRow(rowText, icon) {
  * @property {string} props.tripId The document id associated with the trip.
  * @property {Function} props.handleEditTrip Event handler responsible for
  *     displaying the edit trip modal.
+ * @property {TripView} props.tripView The current user's trips page view.
+ * @property {string} props.eventKey ...
```
Comment (ananyeet) [inline frontend/src/components/ViewTrips/trip.js:78]: `...`  description.

File: frontend/src/components/ViewTrips/change-collaborator-type-button.js (hunk 1 above)
Comment (zghera) [reply frontend/src/components/ViewTrips/change-collaborator-type-button.js:47]: Fixed.

File: frontend/src/components/ViewTrips/trip.js (hunk 2 above)
Comment (zghera) [reply frontend/src/components/ViewTrips/trip.js]: Merged up the PR chain so this is fixed now.

**Comment by zghera** [comment]
Received @Hiram-Silvey 's verbal approval to merge since this is the last day of the internship and he was not able to review the PR.

**Pull request merged**